receiver/dockerstatsreceiver/receiver_test.go

```diff
@@ -54,19 +62,147 @@ func TestErrorsInStart(t *testing.T) {
+	defer singleContainerEngineMock.Close()
+
+	twoContainerEnginerMock, err := dockerMockServer(&map[string]string{
+		"/v1.22/containers/json":                          filepath.Join(mockFolder, "two_containers", "containers.json"),
+		"/v1.22/containers/" + containerIDs[1] + "/json":  filepath.Join(mockFolder, "two_containers", "container1.json"),
+		"/v1.22/containers/" + containerIDs[2] + "/json":  filepath.Join(mockFolder, "two_containers", "container2.json"),
+		"/v1.22/containers/" + containerIDs[1] + "/stats": filepath.Join(mockFolder, "two_containers", "stats1.json"),
+		"/v1.22/containers/" + containerIDs[2] + "/stats": filepath.Join(mockFolder, "two_containers", "stats2.json"),
+	})
+	assert.NoError(t, err)
+	defer twoContainerEnginerMock.Close()
+
+	testCases := []struct {
+		desc                string
+		scrape              func(*receiver) (pmetric.Metrics, error)
+		expectedMetricsFile string
+		mockDockerEngine    *httptest.Server
+	}{
+		{
+			desc: "scrapeV1_single_container",
+			scrape: func(rcv *receiver) (pmetric.Metrics, error) {
+				return rcv.scrape(context.Background())
+			},
+			expectedMetricsFile: filepath.Join(mockFolder, "single_container", "expected_metrics.json"),
+			mockDockerEngine:    singleContainerEngineMock,
+		},
+		{
+			desc: "scrapeV2_single_container",
+			scrape: func(rcv *receiver) (pmetric.Metrics, error) {
+				return rcv.scrapeV2(context.Background())
+			},
+			expectedMetricsFile: filepath.Join(mockFolder, "single_container", "expected_metrics.json"),
+			mockDockerEngine:    singleContainerEngineMock,
+		},
+		{
+			desc: "scrapeV1_two_containers",
+			scrape: func(rcv *receiver) (pmetric.Metrics, error) {
+				return rcv.scrape(context.Background())
+			},
+			expectedMetricsFile: filepath.Join(mockFolder, "two_containers", "expected_metrics.json"),
+			mockDockerEngine:    twoContainerEnginerMock,
+		},
+		{
+			desc: "scrapeV2_two_containers",
+			scrape: func(rcv *receiver) (pmetric.Metrics, error) {
+				return rcv.scrapeV2(context.Background())
+			},
+			expectedMetricsFile: filepath.Join(mockFolder, "two_containers", "expected_metrics.json"),
+			mockDockerEngine:    twoContainerEnginerMock,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			cfg := createDefaultConfig().(*Config)
+			cfg.Endpoint = tc.mockDockerEngine.URL
+			cfg.EnvVarsToMetricLabels = map[string]string{"ENV_VAR": "env-var-metric-label"}
+			cfg.ContainerLabelsToMetricLabels = map[string]string{"container.label": "container-metric-label"}
+
+			receiver := newReceiver(componenttest.NewNopReceiverCreateSettings(), cfg)
+			err := receiver.start(context.Background(), componenttest.NewNopHost())
+			require.NoError(t, err)
+
+			actualMetrics, err := tc.scrape(receiver)
+			require.NoError(t, err)
+
+			expectedMetrics, err := golden.ReadMetrics(tc.expectedMetricsFile)
+
+			// Unset various fields for comparison purposes (non-mdatagen implementation doesn't have these set)
```

Comment:
Can this be broken out and only invoked within the v2 scrape methods instead of all tests?

Reply:
Done!